pytest-embedded-qemu/pytest_embedded_qemu/qemu.py

```diff
@@ -87,6 +144,47 @@ def __init__(
             **kwargs,
         )
 
+    def execute_efuse_command(self, command: str):
+        import espefuse
+        import pexpect
+
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            s.bind(('127.0.0.1', 0))
+            _, available_port = s.getsockname()
+
+        run_qemu_command = [
+            '-nographic',
+            '-machine',
+            self.app.target,
+            '-drive',
+            f'file={self.image_path},if=mtd,format=raw',
+            '-global',
+            f'driver={self.app.target}.gpio,property=strap_mode,value=0x0f',
+            '-drive',
+            f'file={self.efuse_path},if=none,format=raw,id=efuse',
+            '-global',
+            f'driver=nvram.{self.app.target}.efuse,property=drive,value=efuse',
+            '-serial',
+            f'tcp::{available_port},server,nowait',
```

Comment:
When launching QEMU and expecting that it should communicate with another application, it is better to use `-daemonize` flag — as recommended by QEMU documentation. Otherwise there may be a race condition between QEMU getting ready to accept TCP connections and the launch of the other application. It seems `child.expect('qemu')` aims to work around that, but I am not sure this is reliable enough.

Reply:
Also, now that I'm thinking of this... don't you end up with two QEMU processes running at the same time, and accessing the same eFuse file?.. One QEMU process is already launched when `dut` is created, another one is launched here.